analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/container/analysis/sensor/SensorsExecutor.java

```diff
@@ -59,9 +59,19 @@ public SensorsExecutor(DefaultSensorContext context, SensorOptimizer sensorOptim
   }
 
   public void execute() {
-    var sensorGroups = sensors.stream().collect(Collectors.partitioningBy(Sensor.class::isInstance));
-    var moduleSensors = sensorGroups.get(true);
-    var globalSensors = sensorGroups.get(false);
+    var sensorGroups = sensors.stream().collect(Collectors.partitioningBy(s -> {
+      var isModernGlobalSensor = !(s instanceof Sensor);
+
+      var descriptor = new DefaultSensorDescriptor();
+      s.describe(descriptor);
```

Comment:
You can save the call to this if you already know `isModernGlobalSensor` is true